### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by klausenbusk**
I haven't got time to test this (yet). Feel free to test it!

### CODE REVIEW

**Comment by klausenbusk** [comment]
I just did a force push, which should fix the `shellcheck` errors.

File: http/install-common.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+set -x
+
+ln -sf /usr/share/zoneinfo/UTC /etc/localtime
+sed -i -e 's/^#\(en_US.UTF-8\)/\1/' /etc/locale.gen
+locale-gen
+echo 'LANG=en_US.UTF-8' >/etc/locale.conf
+
+# setting the user credentials
+echo -e "${NEWUSER}\n${NEWUSER}" | passwd
+useradd -m -U vagrant
```
Comment (shibumi) [inline http/install-common.sh]: You add a vagrant user in the common file, this is not needed for cloud builds (because we build no vagrant images there).

Comment (klausenbusk) [reply http/install-common.sh]: I missed that line. Fixed

**Comment by shibumi** [comment]
Awesome, looks good. Can you 'squash' all commits to one commit? this makes reverting easier in case something bad happens. I will do some tests afterwards.

**Comment by klausenbusk** [comment]
> Awesome, looks good. Can you 'squash' all commits to one commit? this makes reverting easier in case something bad happens. I will do some tests afterwards.

Personally I prefer [atomic commits](https://www.freshconsulting.com/atomic-commits/) and if I squash the commits, we end up with a big commit which edit a lot of unrelated things (+ reverting a merge is easy: https://mijingo.com/blog/reverting-a-git-merge).

You decide :)

**Comment by klausenbusk** [comment]
So I just got time to do a test build and it didn't build, so I just did a force push with a fix:
```diff
diff --git a/cloud.json b/cloud.json
index 0172bab..2426926 100644
--- a/cloud.json
+++ b/cloud.json
@@ -39,7 +39,7 @@
             ],
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-cloud}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-cloud.sh && systemctl reboot<enter>"
             ]
         }
diff --git a/local.json b/local.json
index 0367dd9..d133066 100644
--- a/local.json
+++ b/local.json
@@ -44,7 +44,7 @@
             ],
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-chroot}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-chroot.sh && systemctl reboot<enter>"
             ]
         }, {
@@ -73,7 +73,7 @@
             ],
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-chroot}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-chroot.sh && systemctl reboot<enter>"
             ]
         }, {
@@ -94,7 +94,7 @@
             "cpus": "{{user `cpus`}}",
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-chroot}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-chroot.sh && systemctl reboot<enter>"
             ]
         }
diff --git a/vagrant.json b/vagrant.json
index 1cbfa5d..f4edad5 100644
--- a/vagrant.json
+++ b/vagrant.json
@@ -44,7 +44,7 @@
             ],
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-chroot}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-chroot.sh && systemctl reboot<enter>"
             ]
         }, {
@@ -73,7 +73,7 @@
             ],
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-chroot}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-chroot.sh && systemctl reboot<enter>"
             ]
         }, {
@@ -94,7 +94,7 @@
             "cpus": "{{user `cpus`}}",
             "boot_command": [
                 "<enter><wait10><wait10><wait10><wait10><wait10><enter><enter>",
-                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,common,-chroot}.sh'<enter><wait>",
+                "curl -O 'http://{{.HTTPIP}}:{{.HTTPPort}}/install{,-{common,cloud}}.sh'<enter><wait>",
                 "bash install.sh < install-chroot.sh && systemctl reboot<enter>"
             ]
         }
```

**Comment by klausenbusk** [comment]
> So I just got time to do a test build and it didn't build, so I just did a force push with a fix:

.. and it takes forever to build, so I stumbled on another thing which is broken.

**Comment by klausenbusk** [comment]
It should work now. See: https://github.com/archlinux/arch-boxes/issues/30#issuecomment-590141765

**Pull request merged**